onnxruntime/core/framework/session_options.h

```diff
@@ -184,6 +187,18 @@ struct SessionOptions {
   // User specified logging func and param
   OrtLoggingFunction user_logging_function = nullptr;
   void* user_logging_param = nullptr;
+
+  // Load cancellation flag is necessary to be within shared memory as session_options are
+  // copied internally and the flag needs to be accessible across all copies.
+  std::shared_ptr<std::atomic_bool> load_cancellation_flag = std::make_shared<std::atomic_bool>(false);
```

Comment:
nit: since `load_cancellation_flag` is public, also having `SetLoadCancellationFlag()` and `IsLoadCancellationFlagSet()` seems redundant.

should I call the accessors or just access the data member directly? the former is done in most places and the latter is done here:
https://github.com/microsoft/onnxruntime/blob/26b11664c8af91537cdc7147586a172ffb874ce6/onnxruntime/python/onnxruntime_pybind_state.cc#L1759

consider making the `load_cancellation_flag` private and requiring use of the accessors, or make it `const std::shared_ptr<std::atomic_bool>` so users can read and write the pointed-to value but not modify the shared_ptr.

Reply:
I am reluctant to add private fields to a structure that has always been public